src/Codeception/Lib/ParamsLoader.php

```diff
@@ -115,7 +115,14 @@ protected function loadXmlFile()
     protected function loadDotEnvFile()
     {
         if (class_exists('Dotenv\Dotenv')) {
-            if (method_exists('Dotenv\Dotenv', 'create')) {
+            if (class_exists('Dotenv\Repository\RepositoryBuilder')) {
+                //dotenv v4
+                $repository = \Dotenv\Repository\RepositoryBuilder::create()
+                    ->withReaders([new \Dotenv\Repository\Adapter\ServerConstAdapter()])
+                    ->immutable()
+                    ->make();
+                $dotEnv = \Dotenv\Dotenv::create($repository, codecept_root_dir(), $this->paramStorage);
+            } elseif (method_exists('Dotenv\Dotenv', 'create')) {
```

Comment:
Backwards compatibility code with v3 and v2 was left intact below, but lowest supported version was raised to 4.0.
Usually we prefer to support as many versions as possible, because you can never know if some app is tied to specific version of a library.
Please change version constraint to `"^2.0 | ^3.0 | ^4.0"` or `">=2.0 <5.0"` or, if you have a good reason, remove initialization code of unsupported versions.

Reply:
Thanks

Reply:
Yes, it makes sense include all the supported versions in version constraint 👍